test/proxy_spec.js

```diff
@@ -513,3 +528,84 @@ describe("Proxy Tests", function () {
       .then(done);
   });
 });
+
+describe("Proxy Tests with Unix socket", function () {
+  var listenOptions = {
+    socket: "/tmp/test.sock",
+  };
+  var requestOptions = new URL("http://localhost");
+  requestOptions.socketPath = listenOptions.socket;
+  var proxy;
+
+  beforeEach(function (callback) {
+    util.setupProxy(listenOptions).then(function (newProxy) {
+      proxy = newProxy;
+      callback();
+    });
+  });
+
+  afterEach(function (callback) {
+    util.teardownServers(callback);
+  });
+
+  it("unix socket HTTP request", function (done) {
+    http
```

Comment:
Using `http.request` because I couldn't get the `fetch` API to work with a unix socket.